### PR TITLE
Bratseth/demand fewer metric readings

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java [hunk 1]
```diff
@@ -159,10 +160,15 @@ static Duration maxScalingWindow() {
...
-        return 4;
+    /** Returns the minimum measurements per node (average) we require to give autoscaling advice.*/
+    private int minimumMeasurementsPerNode(Duration scalingWindow) {
+        // Measurements are ideally taken every minute, but no guarantees
+        // (network, nodes may be down, collecting is single threaded and may take longer than 1 minute to complete.
+        // Since the metric window is 5 minutes, we won't really improve from measuring more often:
+        long minimumMeasurements = scalingWindow.toMinutes() / 5;
+        minimumMeasurements = Math.round(0.8 * minimumMeasurements); // Allow 20% metrics collection blackout
+        if (minimumMeasurements < 1) minimumMeasurements = 1;
+        return (int)minimumMeasurements;
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java:171]: Consider simplifying to `Math.max(1, minimumMeasurements)`.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java [hunk 2]
```diff
@@ -159,10 +160,15 @@ static Duration maxScalingWindow() {
...
     }
 
-    /** Measurements are currently taken once a minute. See also scalingWindow */
-    static int minimumMeasurementsPerNode(ClusterSpec cluster) {
-        if (cluster.isStateful()) return 60;
-        return 4;
+    /** Returns the minimum measurements per node (average) we require to give autoscaling advice.*/
+    private int minimumMeasurementsPerNode(Duration scalingWindow) {
+        // Measurements are ideally taken every minute, but no guarantees
+        // (network, nodes may be down, collecting is single threaded and may take longer than 1 minute to complete.
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java:166]: Unmatched `(`. 😄

**Review by mpolden — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java (hunk 2 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java:166]: Thanks. i'll fix it later

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java (hunk 1 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java:171]: I always have to think for 1 second when I see max implementing min, so some times I try to do without.

**Pull request merged**